src/redux/middleware/intialStartUpMiddleware.ts

```diff
@@ -1,28 +1,24 @@
-import { AnyAction, Middleware } from 'redux'
+import { Middleware, AnyAction } from 'redux'
+import { BOOTSTRAP } from '../actions'
 import { PriceSingleDataPoint } from '../../models/prices'
 import { CompanyOverview } from '../../features/companyOverview/models/companyOverview'
 import { News } from '../../features/news/models/news'
 import { KeyStats } from '../../features/keystats/models/keyStats'
 import { ChartSingleDataPoint } from '../../features/charts/models'
-import { Error } from '../../models/errors'
+import { StockListItem } from 'features/search/models'
+import { updateStockList } from 'features/search/redux'
 import { 
     updatePricesData,
     updateChartData,
-    updateChartRange, UPDATE_CHART_RANGE,
-    resetState,
     updateCompany,
     updateKeyStats,
     updateNews,
     errorAction,
-    UPDATE_TICKER
 } from '../actions';
-import { StockListItem } from 'features/search/models'
-import { updateStockList } from 'features/search/redux'
-
 
-const socketMiddleware = (socket: SocketIOClient.Socket, defaultTicker: string = 'aapl'): Middleware => {
 
-    return ({ dispatch, getState }) => {
+export const initialStartUpMiddleware = (socket: SocketIOClient.Socket, defaultTicker: string = 'aapl'): Middleware => {
```

Comment:
I like using a `dependencies` object so it is easy to provide additional deps to middleware.

```ts
type Dependencies = { socket: SocketIOClient.Socket; defaultTicker?: string; };

const middleware = ({ socket, defaultTicker }: Dependencies): Middleware<{}, AppState> => ...
```

Reply:
I guess this is better suited for larger scale applications where the middleware utilises more deps.

Anyway, I'm always a big fan of grouping things together for a cleaner code base so it will be actioned ad updated in next push.